third_party/benchmarks/python/pystone.py

```diff
@@ -0,0 +1,247 @@
+"""
+"PYSTONE" Benchmark Program
+
+Version:        Python/1.1 (corresponds to C/1.1 plus 2 Pystone fixes)
+
+Author:         Reinhold P. Weicker,  CACM Vol 27, No 10, 10/84 pg. 1013.
+
+                Translated from ADA to C by Rick Richardson.
+                Every method to preserve ADA-likeness has been used,
+                at the expense of C-ness.
+
+                Translated from C to Python by Guido van Rossum.
+
+Version History:
+
+                Version 1.1 corrects two bugs in version 1.0:
+
+                First, it leaked memory: in Proc1(), NextRecord ends
+                up having a pointer to itself.  I have corrected this
+                by zapping NextRecord.PtrComp at the end of Proc1().
+
+                Second, Proc3() used the operator != to compare a
+                record to None.  This is rather inefficient and not
+                true to the intention of the original benchmark (where
+                a pointer comparison to None is intended; the !=
+                operator attempts to find a method __cmp__ to do value
+                comparison of the record).  Version 1.1 runs 5-10
+                percent faster than version 1.0, so benchmark figures
+                of different versions can't be compared directly.
+"""
+
+# Support running on MicroPython added by Toit.
+
+import sys
+if sys.implementation.name == 'micropython':
+  import utime
+  def time_ms():
+    return utime.ticks_ms()
+else:
+  import time
+  def time_ms():
+    return int(time.time() * 1000)
+
+LOOPS = 10000
+
+[Ident1, Ident2, Ident3, Ident4, Ident5] = range(1, 6)
+
+class Record:
+
+    def __init__(self, PtrComp = None, Discr = 0, EnumComp = 0,
+                       IntComp = 0, StringComp = 0):
+        self.PtrComp = PtrComp
+        self.Discr = Discr
+        self.EnumComp = EnumComp
+        self.IntComp = IntComp
+        self.StringComp = StringComp
+
+    def copy(self):
+        return Record(self.PtrComp, self.Discr, self.EnumComp,
+                      self.IntComp, self.StringComp)
+
+TRUE = 1
+FALSE = 0
+
+def main(iterations):
+    startTime = time_ms()
+    for i in range(iterations):
+      pystones(LOOPS)
+    endTime = time_ms()
+    ms = (endTime - startTime) / iterations
+    print("Pystone (%s) - time per iteration: %i ms" % (sys.implementation.name, ms))
```

Comment:
If it's a cut down version we should not that right in the print statement.  So easy to forget a few years down the line and wonder why Python is so fast.

Reply:
Was this comment addressed?